crates/wac-graph/src/encoding.rs

```diff
@@ -325,6 +325,7 @@ impl<'a> TypeEncoder<'a> {
                 let index = state.current.encodable.type_count();
                 state.current.encodable.ty().instance(&ty);
                 log::debug!("instance {id} encoded to type index {index}");
+                state.current.instances.insert(id, index);
```

Comment:
I don't believe this is correct; this is encoding an _instance type_ and not an _instance_ (i.e. the index is for the type index space not for the instance index space).

The `TypeEncoder` concerns itself with encoding types, but certain encodings (such as interfaces, worlds, and component types) may contain imports of _instances_ for expressing the dependency on "used" interfaces, which are kept track of in the `instances` field on `Scope`. It is expected that `instances` only stores indexes into the instances index space.

Reply:
The log message should really be `log::debug!("interface {id} encoded to type index {index}");`, though; "instance" here is misleading.